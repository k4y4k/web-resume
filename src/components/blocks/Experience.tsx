import { graphql, useStaticQuery } from 'gatsby'
import * as React from 'react'
import ItemContainer from '../section/ItemContainer'
import SectionContainer from '../section/SectionContainer'

interface ExperienceItem {
  company: string
  endDate: string
  highlights?: string[]
  position: string
  startDate: string
  summary: string
}

interface two {
  history: ExperienceItem[]
}

export const PureExperience = ({ history }: two): JSX.Element => {
  return (
    <div data-testid='experience'>
      <SectionContainer title='Experience'>
        {history.map((el, i: number) => {
          return (
            <ItemContainer
              key={i}
              details={el.summary}
              fromDate={el.startDate}
              toDate={el.endDate}
              title={el.position}
              subtitle={el.company}
              highlights={el.highlights}
            />
          )
        })}
      </SectionContainer>
    </div>
  )
}

const Experience = (): JSX.Element => {
  const data = useStaticQuery(graphql`
    {
      file(name: { eq: "data" }, extension: { eq: "json" }) {
        id
        childDataJson {
          work {
            company
            endDate
            highlights
            position
            startDate
            summary
          }
        }
      }
    }
  `)

  const { work } = data?.file.childDataJson

  return <PureExperience history={work} />
}

export default Experience
