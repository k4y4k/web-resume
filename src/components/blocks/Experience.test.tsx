import { screen, render } from '@testing-library/react'
import * as React from 'react'
import { PureExperience as Experience } from './Experience'

describe('<Experience />', () => {
  test('bails out on no data', () => {
    render(<Experience history={[]} />)

    // still want the title
    expect(screen.getByTestId('experience')).toBeInTheDocument()

    expect(screen.queryByTestId('sectionItemContainer')).not.toBeInTheDocument()
  })

  test('renders array of >=1 experience items', () => {
    const data = [
      {
        company: 'Company',
        position: 'President',
        website: 'http://company.com',
        startDate: '2013-01-01',
        endDate: '2014-01-01',
        summary: 'Description...',
        highlights: ['Started the company'],
      },
    ]

    render(<Experience history={data} />)

    const experienceList = screen.getAllByTestId('experience')

    expect(experienceList.length).toBe(1)
  })
})
