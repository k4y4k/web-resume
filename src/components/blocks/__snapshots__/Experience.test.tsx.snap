// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience /> renders OK 1`] = `
<div
  data-testid="experience"
>
  <div
    data-testid="sectionContainer"
  >
    <h2
      data-testid="sectionTitle"
    >
      Experience
    </h2>
    <div
      data-testid="sectionItemContainer"
    >
      <h1
        data-testid="itemTitle"
      >
        Volunteer
      </h1>
      <div
        data-testid="byline"
      >
        <span
          data-testid="itemSubtitle"
        >
          Organization
        </span>
        |
        <p
          data-testid="sectionItemDates"
        >
          Mar 2016
           - 
          Feb 2017
        </p>
      </div>
      <div
        data-testid="sectionItemDetails"
      >
        <p>
          Description...
        </p>
      </div>
    </div>
    <div
      data-testid="sectionItemContainer"
    >
      <h1
        data-testid="itemTitle"
      >
        President
      </h1>
      <div
        data-testid="byline"
      >
        <span
          data-testid="itemSubtitle"
        >
          Company
        </span>
        |
        <p
          data-testid="sectionItemDates"
        >
          2013
           - 
          2014
        </p>
      </div>
      <div
        data-testid="sectionItemDetails"
      >
        <p>
          Description...
        </p>
      </div>
    </div>
    <div
      data-testid="sectionItemContainer"
    >
      <h1
        data-testid="itemTitle"
      >
        Volunteer
      </h1>
      <div
        data-testid="byline"
      >
        <span
          data-testid="itemSubtitle"
        >
          Organization
        </span>
        |
        <p
          data-testid="sectionItemDates"
        >
          2012
           - 
          2013
        </p>
      </div>
      <div
        data-testid="sectionItemDetails"
      >
        <p>
          Description...
        </p>
      </div>
    </div>
  </div>
</div>
`;
